tower-batch/src/service.rs

```diff
@@ -73,8 +99,28 @@ where
         T::Error: Send + Sync,
         Request: Send + 'static,
     {
-        let (batch, worker) = Self::pair(service, max_items, max_latency);
-        tokio::spawn(worker.run());
+        let (mut batch, worker) = Self::pair(service, max_items, max_latency);
+
+        let span = info_span!("batch worker", kind = std::any::type_name::<T>());
+
+        #[cfg(tokio_unstable)]
+        let worker_handle = {
+            let batch_kind = std::any::type_name::<T>();
+
+            // TODO: identify the unique part of the type name generically,
+            //       or make it an argument to this method
+            let batch_kind = batch_kind.trim_start_matches("zebra_consensus::primitives::");
+            let batch_kind = batch_kind.trim_end_matches("::Verifier");
+
+            tokio::task::Builder::new()
+                .name(&format!("{} batch", batch_kind))
+                .spawn(worker.run().instrument(span))
```

Comment:
Bug: five days before this PR was merged, `Builder::spawn` was made fallible in upstream (https://github.com/tokio-rs/tokio/pull/4823). This code has not compiled for the past several years.